crates/payload/primitives/src/payload.rs

```diff
@@ -164,3 +167,28 @@ impl ExecutionPayload for op_alloy_rpc_types_engine::OpExecutionData {
         self.payload.as_v1().gas_used
     }
 }
+
+/// Special implementation for Ethereum types that provides additional helper methods
+impl<'a, Attributes> PayloadOrAttributes<'a, ExecutionData, Attributes>
+where
+    Attributes: PayloadAttributes,
+{
+    /// Return the execution requests from the payload, if available.
+    ///
+    /// This will return `Some(requests)` only if:
+    /// - The payload is an `ExecutionData` (not `PayloadAttributes`)
+    /// - The payload has Prague payload fields
+    /// - The Prague fields contain requests (not a hash)
+    ///
+    /// Returns `None` in all other cases.
+    pub fn execution_requests(&self) -> Option<&Requests> {
+        if let Self::ExecutionPayload(payload) = self {
+            payload.sidecar.prague().and_then(|prague_fields| match &prague_fields.requests {
```

Comment:
there's also fn requests -> Option<&Requests> I believe

Reply:
awesome thx! changes to use it here https://github.com/paradigmxyz/reth/pull/14932/commits/3cc6d1b160114bfea103a011ac5e78b1340e35ec